contracts/SinglePlayerCommit.sol

```diff
@@ -163,84 +149,72 @@ contract SinglePlayerCommit is ChainlinkClient, Ownable {
     /// @notice Create commitment, store on-chain and emit event
     /// @param _activityKey Keccak256 hashed, encoded name of activity
     /// @param _goalValue Distance of activity as goal
-    /// @param _startTime Starttime of commitment, also used for endTime
+    /// @param _daysToStart Starttime of commitment, also used for endTime
     /// @param _stake Amount of <token> to stake againt achieving goal
     /// @param _userId ???
     /// @dev Check parameters, create commitment, store on-chain and emit event
     function makeCommitment(
         bytes32 _activityKey,
         uint256 _goalValue,
-        uint256 _startTime,
+        uint256 _daysToStart,
+        uint256 _amountOfDays,
         uint256 _stake,
         string memory _userId
-    ) 
-        public 
-        returns (bool success) 
-    {
+    ) public returns (bool success) {
         console.log("makeCommitment called by %s", msg.sender);
 
         require(!commitments[msg.sender].exists, "SPC::makeCommitment - msg.sender already has a commitment");
         require(activities[_activityKey].allowed, "SPC::makeCommitment - activity doesn't exist or isn't allowed");
-        require(_startTime > block.timestamp, "SPC::makeCommitment - commitment cannot start in the past");
         require(_goalValue > 1, "SPC::makeCommitment - goal is too low");
         require(committerBalances[msg.sender] >= _stake, "SPC::makeCommitment - insufficient token balance");
 
-        uint256 endTime = _startTime.add(7 days);
-
-        Commitment memory commitment = Commitment({
-            committer: msg.sender,
-            activityKey: _activityKey,
-            goalValue: _goalValue,
-            startTime: _startTime,
-            endTime: endTime,
-            stake: _stake,
-            reportedValue: 0,
-            lastActivityUpdate: 0,
-            met: false,
-            userId: _userId,
-            exists: true
-        });
+        uint256 startTime = _daysToStart > 0 ? addDays(_daysToStart, block.timestamp): block.timestamp;
```

Comment:
Similarly here, I think we may need to user something other than the addDays function, e.g. if the offset is in hours or fractions of days.